util/pkg/vfs/azureblob.go

```diff
@@ -100,53 +103,44 @@ func (p *AzureBlobPath) Join(relativePath ...string) Path {
 
 // ReadFile returns the content of the blob.
 func (p *AzureBlobPath) ReadFile(ctx context.Context) ([]byte, error) {
-	var b bytes.Buffer
-	_, err := p.WriteTo(&b)
-	if err != nil {
-		return nil, err
-	}
-	return b.Bytes(), nil
-}
-
-// WriteTo writes the content of the blob to the writer.
-func (p *AzureBlobPath) WriteTo(w io.Writer) (n int64, err error) {
-	ctx := context.TODO()
+	klog.V(8).Infof("Reading file: %s - %s", p.container, p.key)
 
 	client, err := p.getClient(ctx)
 	if err != nil {
-		return 0, err
+		return nil, err
 	}
 
-	cURL, err := client.newContainerURL(p.container)
+	get, err := client.DownloadStream(ctx, p.container, p.key, nil)
 	if err != nil {
-		return 0, err
+		if bloberror.HasCode(err, bloberror.ContainerNotFound) || bloberror.HasCode(err, bloberror.BlobNotFound) {
+			return nil, os.ErrNotExist
+		}
+		return nil, err
 	}
-	resp, err := cURL.NewBlockBlobURL(p.key).Download(
-		ctx,
-		0, /* offset */
-		azblob.CountToEnd,
-		azblob.BlobAccessConditions{},
-		false, /* rangeGetContentMD5 */
-		azblob.ClientProvidedKeyOptions{},
-	)
+
+	b := &bytes.Buffer{}
+	retryReader := get.NewRetryReader(ctx, &azblob.RetryReaderOptions{})
+	_, err = b.ReadFrom(retryReader)
 	if err != nil {
-		serr, ok := err.(azblob.StorageError)
-		if ok && serr.ServiceCode() == azblob.ServiceCodeBlobNotFound {
-			return 0, os.ErrNotExist
-		}
-		return 0, err
+		return nil, err
 	}
-	return io.Copy(w, resp.Body(azblob.RetryReaderOptions{MaxRetryRequests: 10}))
+
+	return b.Bytes(), nil
+}
+
+// WriteTo writes the content of the blob to the writer.
+func (p *AzureBlobPath) WriteTo(w io.Writer) (n int64, err error) {
+	return 0, fmt.Errorf("not implemented")
```

Comment:
any reason not to implement this? it was implemented before

Reply:
It is used only in etcd-manager, not sure why.
I added it back for now. Thanks for the reminder.